pkgs/by-name/ve/vep/package.nix

```diff
@@ -0,0 +1,122 @@
+    stdenv.mkDerivation {
+      inherit name version;
+      # Not available through fetchFromGitHub
+      src = fetchzip {
+        inherit sha256 version;
+        url = "https://github.com/Ensembl/${name}/archive/release/${version}.zip";
+      };
+      installPhase = customInstallPhase;
+    };
+
+  vepPlugins = fetchFromGitHub {
+    owner = "Ensembl";
+    repo = "VEP_plugins";
+    rev = "8f271c4848338dc7d504881ff71fdf2892c3d096";
+    sha256 = "sha256-LbaXwLFDP3m1QhRHwO9uh36BEFHE2NzL4xdxTb7/S5Q=";
+  };
+
+  # Install ensembl-xs, faster run using re-implementation in C of some of the Perl subroutines
+  ensembl-xs = perlPackages.buildPerlPackage rec {
+    pname = "ensembl-xs";
+    version = "2.3.2";
+    src = fetchFromGitHub {
+      inherit version;
+      owner = "Ensembl";
+      repo = "ensembl-xs";
+      rev = version;
+      sha256 = "1qqnski532f4bz32wxbqd9w1sz40rjh81ipp9p02k3rlaf1gp1fa";
+    };
+    # PREFIX is important
+    configurePhase = ''
+      perl Makefile.PL PREFIX=$out INSTALLDIRS=site
+    '';
+    # Test do not work -- wrong include path
+    doCheck = false;
+  };
+
+  # it contains compiled versions of certain key subroutines used in VEP
+  ensembl = ensemblGit "ensembl" "sha256-labyyqp+FRLIN9y4koePkX5fUls8HbCJlgYHUA0JR/0=";
+  ensembl-io = ensemblGit "ensembl-io" "sha256-r3RvN5U2kcyghoKM0XuiBRe54t1U4FaZ0QEeYIFiG0w=";
+  ensembl-variation = ensemblGit "ensembl-variation" "sha256-UjrLHF9EqI+Mp+SZR4sLNZUCGiA/UYhoFWtpwiKF8tM=";
+  ensembl-funcgen = ensemblGit "ensembl-funcgen" "sha256-a9hxLBoXJsF5JWuRdpyOac1u033M8ivEjEQecuncghs=";
+in
+perlPackages.buildPerlModule rec {
+  inherit version;
+  pname = "ensembl-vep";
+  buildInputs =
+    (with perlPackages; [
+      ArchiveZip
+      BioBigFile
+      BioDBHTS
+      BioExtAlign
+      BioPerl
+      DBI
+      DBDmysql
+      LWP
+      JSON
+    ])
+    ++ [
+      ensembl-xs
+      ensembl
+      ensembl-funcgen
+      ensembl-io
+      ensembl-variation
+    ];
+  propagatedBuildInputs = [ htslib ];
+  src = fetchFromGitHub {
+    owner = "Ensembl";
+    repo = pname;
+    rev = "release/${version}";
+    sha256 = "sha256-6lRdWV2ispl+mpBhkZez/d9PxOw1fkNUWeG8mUIqBJc=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  dontBuild = true;
+  doCheck = false;
+
+  outputs = [ "out" ];
+  postFixup = ''
+    wrapProgram $out/bin/vep \
+      --add-flags "--dir_plugins ${vepPlugins}";
+  '';
+  installPhase = ''
+    mkdir -p $out/bin
+    cp vep filter_vep haplo variant_recoder $out/bin/
+
+    mkdir -p $out/${perl.libPrefix}/${perl.version}/
+    cp -r modules/Bio $out/${perl.libPrefix}/${perl.version}/
+  '';
+
+  meta = {
+    homepage = "https://www.ensembl.org/info/docs/tools/vep/index.html";
+    description = "Annotate genetics variants based on genes, transcripts, and protein sequence, as well as regulatory regions";
+    license = lib.licenses.asl20;
+    mainProgram = "vep";
+    maintainers = with lib.maintainers; [ apraga ];
```

Comment:
You should add a [`platforms`](https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms) attribute.